app/controllers/concerns/hashable_params.rb

```diff
@@ -8,7 +8,7 @@ module HashableParams
   def params_to_hash(input_params)
     return {} if input_params.blank?
     if rails5?
-      input_params.to_unsafe_h
+      input_params.try(:to_unsafe_h) || input_params
```

Comment:
Turns out - after running this and the `each_index` fix together - this does sometimes end up with Hash input - so this assumes that's what's happened and returns the unchanged input as the result if it can't call `to_unsafe_h`

Reply:
I think it's advisable to deal with the possibility of it receiving a `Hash` (blank params seem to be passed as an empty Hash so there's that, not entirely sure how we end up passing it populated Hashes now and again but ... having the "just call this, it'll be fine" method deal with everything feels like the right first step).

But... you've helped me realise that we can drop the `if rails5?` check instead and just return `input_params.clone` if the input doesn't respond to `to_unsafe_h`

5918a2e collapses my initial nonsense down to make this a 2 line method

Reply:
Ok so - less lazy answer - aside from where I've written a test that supplies a populated Hash to prove it can cope with it, it genuinely receives a Hash as input when called from `TranslatableParams::WhitelistedParams` and while I _could_ treat that as a special case it leaves me worried that there might be other exceptions involving constructed and/or nested params so I think I'll keep it as the responsibility of the method to Do The Right Thing.

Less satisfying in 43b27144